File: Examples/src/main/java/com/aspose/pdf/examples/AsposePdf/Documents/SetPDFFileInfo.java
/* 
 * Copyright 2001-2015 Aspose Pty Ltd. All Rights Reserved.
 *
 * This file is part of Aspose.Pdf. The source code in this file
 * is only intended as a supplement to the documentation, and is provided
 * "as is", without warranty of any kind, either expressed or implied.
 */
 
package com.aspose.pdf.examples.AsposePdf.Documents;

import com.aspose.pdf.examples.Utils;

public class SetPDFFileInfo
{
    public static void main(String[] args) throws Exception
    {
        // The path to the documents directory.
        String dataDir = Utils.getDataDir(SetPDFFileInfo.class);

        //open document
        com.aspose.pdf.Document pdfDocument = new com.aspose.pdf.Document(dataDir+ "input.pdf");
        //get document information
        com.aspose.pdf.DocumentInfo docInfo = pdfDocument.getInfo();
        //set Author information
        docInfo.setAuthor("Aspose.Pdf for java");
        docInfo.setCreationDate(new java.util.Date());
        docInfo.setKeywords("Aspose.Pdf, DOM, API");
        docInfo.setModDate(new java.util.Date());
        docInfo.setSubject("PDF Information");
        docInfo.setTitle("Setting PDF Document Information");
        // save update document with new information
        pdfDocument.save(dataDir+ "Updated_Information.pdf");
        // Display result.
        System.out.println("Process completed successfully!");
    }
}

File: plugins/Aspose_Pdf_Java_for_Ruby/lib/asposepdfjava/Document/setpdffileinfo.rb
module Asposepdfjava
  module SetPdfFileInfo
    def run
    	# The path to the documents directory.
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'
        
        # Open a pdf document.
        doc = Asposepdfjava.java_class('com.aspose.pdf.Document').new(data_dir + "input1.pdf")

        # Get document information
        doc_info = doc.getInfo()
        
        doc_info.setAuthor("Aspose.Pdf for java")
        doc_info.setCreationDate(Asposepdfjava.java_class('java.util.Date').new)
        doc_info.setKeywords("Aspose.Pdf, DOM, API")
        doc_info.setModDate(Asposepdfjava.java_class('java.util.Date').new)
        doc_info.setSubject("PDF Information")
        doc_info.setTitle("Setting PDF Document Information")
        
        # save update document with new information
        doc.save(data_dir + "Updated_Information.pdf")

        puts "Update document information, please check output file."
    end
  end
end
